src/Module.Server/Modes/Dtv/CrpgDtvClient.cs

```diff
@@ -150,4 +153,22 @@ private void HandleViscountUnderAttack(CrpgDtvViscountUnderAttackMessage message
             SoundEventPath = "event:/ui/notification/alert",
         });
     }
+
+    private void HandleServerEventSetStonePileAmmo(SetStonePileAmmo message)
+    {
+        if (message.AmmoCount > 0)
+        {
+            StonePile? stonePile = Mission.MissionNetworkHelper.GetMissionObjectFromMissionObjectId(message.StonePileId) as StonePile;
+            stonePile?.Activate();
+        }
+    }
+
+    private void HandleServerSetRangedSiegeWeaponAmmo(SetRangedSiegeWeaponAmmo message)
+    {
+        if (message.AmmoCount > 0)
+        {
+            RangedSiegeWeapon? rangedSiegeWeapon = Mission.MissionNetworkHelper.GetMissionObjectFromMissionObjectId(message.RangedSiegeWeaponId) as RangedSiegeWeapon;
+            rangedSiegeWeapon?.Activate();
+        }
```

Comment:
Is this really needed?
Correct me if i'm wrong but SetStonePileAmmo and SetRangedSiegeWeaponAmmo already has a handler?

Reply:
The existing handlers do not call Activate() which prohibits the client from interacting with the piles even though they are filling back up.